drivers/modem/modem_cellular.c

```diff
@@ -444,7 +445,7 @@ static void modem_cellular_idle_event_handler(struct modem_cellular_data *data,
 
 	switch (evt) {
 	case MODEM_CELLULAR_EVENT_RESUME:
-		if (modem_cellular_gpio_is_enabled(&config->power_gpio)) {
+		if (modem_cellular_gpio_is_enabled(&config->power_gpio) && !config->autostarts) {
 			modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_POWER_ON_PULSE);
 			break;
 		}
```

Comment:
Could you change this to
```
	case MODEM_CELLULAR_EVENT_RESUME:
		if (config->autostarts) {
			modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_AWAIT_POWER_ON);
			break;
		}

		if (modem_cellular_gpio_is_enabled(&config->power_gpio)) {
			modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_POWER_ON_PULSE);
			break;
		}

		...
```

Without this change, the init script is sent immediately, which will not work for modems which require time to start up before they accept commands.

If the sara R5 does not need any time before accepting commands (it should probably at least get a few seconds after power on), you can adjust the startup time down for the sara R5 in the `MODEM_CELLULAR_DEVICE_U_BLOX_SARA_R5`
```
		.startup_time_ms = 10000,                                                          \
```